airflow/contrib/kubernetes/init_container.py

```diff
@@ -0,0 +1,51 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+
+class InitContainer:
```

Comment:
From a design perspective, do you think it could be a good idea to implement a more generic `Container` class instead, that could eventually be used to specify main, sidecar and init containers for pods in a more generic way?

Reply:
I would agree with this, though I also imagine at some point soon we'll want to revamp this entire section to use the python SDK generated by kubernetes (that way we can allow users to submit a k8s pod object if there is a feature we are not offering).

Reply:
I also agree with this. In this PR my aim is just to provide the init-container class to the Kubernetes Pod Operator with the little impact as possible on the existing code. The use of Kubernetes in Airflow has great potentiality and a broader K8S support is certainly desirable.